Tests/Mocks/RubyPactMockServiceStub.swift

```diff
@@ -1,108 +1,163 @@
+        client?.urlProtocol(self, didLoad: stub.responseBody)
+        client?.urlProtocolDidFinishLoading(self)
     }
-  }
+    
+    override func stopLoading() { }
+    
+    override class func canonicalRequest(for request: URLRequest) -> URLRequest { return request }
+    
 }
 
 struct RubyPactMockServiceStub {
-  var cleanStub: VerifiableHttpStub = VerifiableHttpStub(requestStub: RubyMockServiceRequest.cleanInteractions)
-  var setupInteractionsStub: VerifiableHttpStub = VerifiableHttpStub(requestStub: RubyMockServiceRequest.setupInteractions)
-  var verifyInteractionsStub: VerifiableHttpStub = VerifiableHttpStub(requestStub: RubyMockServiceRequest.verifyInteractions)
-  var writePactStub: VerifiableHttpStub = VerifiableHttpStub(requestStub: RubyMockServiceRequest.writePact)
-
-  @discardableResult
-  func clean(responseCode: Int32,
-             response: String) -> RubyPactMockServiceStub {
-    self.cleanStub.stubWithResponse(responseCode: responseCode, response: response)
-    return self
-  }
-
-  func cleanWithError(errorMessage: String) {
-    self.cleanStub.stubWithError(errorMessage: errorMessage)
-  }
-
-  @discardableResult
-  func setupInteractions(responseCode: Int32,
-                         response: String) -> RubyPactMockServiceStub {
-    self.setupInteractionsStub.stubWithResponse(responseCode: responseCode, response: response)
-    return self
-  }
-
-  @discardableResult
-  func verifyInteractions(responseCode: Int32,
-                          response: String) -> RubyPactMockServiceStub {
-    self.verifyInteractionsStub.stubWithResponse(responseCode: responseCode, response: response)
-    return self
-  }
-
-  @discardableResult
-  func writePact(responseCode: Int32,
-                 response: String) -> RubyPactMockServiceStub {
-    self.writePactStub.stubWithResponse(responseCode: responseCode, response: response)
-    return self
-  }
+    var cleanStub: VerifiableHttpStub { StubProtocol.stubs[.cleanInteractions] ?? .init() }
+    var setupInteractionsStub: VerifiableHttpStub { StubProtocol.stubs[.setupInteractions] ?? .init() }
+    var verifyInteractionsStub: VerifiableHttpStub { StubProtocol.stubs[.verifyInteractions] ?? .init() }
+    var writePactStub: VerifiableHttpStub { StubProtocol.stubs[.writePact] ?? .init() }
+    
+    let session: URLSession
+    
+    init() {
+        let configuration = URLSessionConfiguration.ephemeral
+        configuration.protocolClasses = [StubProtocol.self]
+        self.session = URLSession.init(configuration: configuration)
+    }
+    
+    @discardableResult
+    func clean(responseCode: Int,
+               response: String) -> RubyPactMockServiceStub {
+        StubProtocol.stubs[.cleanInteractions] = .init(responseCode: responseCode, response: response)
+        return self
+    }
+    
+    @discardableResult
+    func cleanWithError(errorMessage: String) -> RubyPactMockServiceStub {
```

Comment:
I think you can delete this cleanWithError function. It used to be used in tests, but is no longer.